catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/converter/mgmp/MgmpConverter.java

```diff
@@ -0,0 +1,998 @@
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Associations;
+import ddf.catalog.data.types.Contact;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.DateTime;
+import ddf.catalog.data.types.Location;
+import ddf.catalog.data.types.Media;
+import ddf.catalog.data.types.Topic;
+
+/**
+ * Not thread safe.
+ */
+public class MgmpConverter extends AbstractGmdConverter {
+
+    // TODO i don't know how to output ISR_DATA_QUALITY, maybe deleted as per code review comments
+    @SuppressWarnings("unused")
+    public static final String ISR_DATA_QUALITY = "ext.isr.data-quality";
+
+    // TODO i don't know how to get the data from metacard to output ISR_DATA_QUALITY, maybe deleted as per code review comments
+    @SuppressWarnings("unused")
+    public static final List<String> DATA_QUALITY_LIST = Arrays.asList(
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/MGMP_UsabilityElement",
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/DQ_GriddedDataPositionalAccuracy",
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/DQ_RelativeInternalPositionalAccuracy",
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/DQ_AbsoluteExternalPositionalAccuracy");
+
+    private static final String BOUNDING_BOX_FORMAT = "%.15f";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MgmpConverter.class);
+
+    private static final int MIN_CLOUD_COVERAGE = 0;
+
+    private static final int MAX_CLOUD_COVERAGE = 100;
+
+    private static final String ID_REGEX = "^.{32}$";
+
+    private static final int ID_SEGMENT_1_START = 0;
+
+    private static final int ID_SEGMENT_1_END = 8;
+
+    private static final int ID_SEGMENT_2_START = 8;
+
+    private static final int ID_SEGMENT_2_END = 12;
+
+    private static final int ID_SEGMENT_3_START = 12;
+
+    private static final int ID_SEGMENT_3_END = 16;
+
+    private static final int ID_SEGMENT_4_START = 16;
+
+    private static final int ID_SEGMENT_4_END = 20;
+
+    private static final int ID_SEGMENT_5_START = 20;
+
+    private static final int ID_SEGMENT_5_END = 32;
+
+    private static final String ID_SEPARATOR = "-";
+
+    private Predicate<String> crsFilter = s -> s.matches("^[^:]+:[^:]+$");
+
+    private int geographicElementIndex = 1;
+
+    private int dateElementIndex = 1;
+
+    @Override
+    protected List<String> getXstreamAliases() {
+        return Arrays.asList(GmdConstants.GMD_LOCAL_NAME, MgmpConstants.GMD_METACARD_TYPE_NAME);
+    }
+
+    /**
+     * Builds up the xml paths and values to write.
+     * Order matters!  Paths should be added in the order they must be written.
+     *
+     * @param metacard the source
+     * @return XstreamPathValueTracker containing XML paths and values to write
+     */
+    @Override
+    protected XstreamPathValueTracker buildPaths(MetacardImpl metacard) {
+
+        XstreamPathValueTracker pathValueTracker = new XstreamPathValueTracker();
```

Comment:
It appears that we are passing pathValueTracker and metacard around to a dozen+ private methods. Recommend moving `pathValueTracker` and `metacard` out to class variables to clean up the readability and redundancy.